runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/DataUtil.java

```diff
@@ -159,6 +156,82 @@ public static Iterable concatNonSerPartitions(final Iterable<NonSerializedPartit
     return concatStream.collect(Collectors.toList());
   }
 
+  /**
+   * An iterator that emits objects from {@link InputStream} using the corresponding {@link Coder}.
+   * @param <T> The type of elements.
+   */
+  public static final class InputStreamIterator<T> implements Iterator<T> {
+
+    private final InputStream inputStream;
+    private final Coder<T> coder;
+    private final Optional<Long> limit;
```

Comment:
It would be good to avoid to use `Optional` as variable, because this class is value-based and non-serializable.